sound/soc/sof/ipc4-pcm.c

```diff
@@ -457,17 +457,41 @@ static void ipc4_ssp_dai_config_pcm_params_match(struct snd_sof_dev *sdev, const
 	}
 }
 
+static int ipc4_set_fmt_mask(struct snd_mask *fmt, unsigned int bit_depth)
+{
+	switch (bit_depth) {
+	case 0:
+		/* bit depth not set, ignore */
+		break;
```

Comment:
With current topologies, this is needed. Not sure whether to keep this in. I'll add a link once I have the tplg PR.

Reply:
why would we not set the bit_depth? this seems like a miss

Reply:
@ranj063 https://github.com/thesofproject/sof/pull/6994

Large number of tests will fail without this. If #6994 was a one-off error, I can remove the "case 0" and make it a fatal error.